iroh-net/src/hp/netcheck.rs

```diff
@@ -1,79 +1,51 @@
-    FutureExt,
-};
 use iroh_metrics::{inc, netcheck::NetcheckMetrics};
-use rand::seq::IteratorRandom;
-use tokio::{
-    net::UdpSocket,
-    sync::{self, mpsc, oneshot, RwLock},
-    task::{AbortHandle, JoinSet},
-    time::{self, Duration, Instant},
-};
+use tokio::net::UdpSocket;
+use tokio::sync::{self, mpsc, oneshot};
+use tokio::time::{Duration, Instant};
 use tokio_util::sync::CancellationToken;
 use tracing::{debug, debug_span, error, info, instrument, trace, warn, Instrument};
 
-use crate::{
-    defaults::DEFAULT_DERP_STUN_PORT,
-    net::{interfaces, ip::to_canonical},
-    util::MaybeFuture,
-};
+use crate::defaults::DEFAULT_DERP_STUN_PORT;
+use crate::net::ip::to_canonical;
+use crate::util::CancelOnDrop;
 
-use self::probe::{Probe, ProbePlan, ProbeProto};
+use self::probe::ProbeProto;
 
-use super::{
-    derp::{DerpMap, DerpNode, DerpRegion, UseIpv4, UseIpv6},
-    dns::DNS_RESOLVER,
-    ping::Pinger,
-    portmapper, stun,
-};
+use super::derp::{DerpMap, DerpNode, UseIpv4, UseIpv6};
+use super::dns::DNS_RESOLVER;
+use super::portmapper;
+use super::stun;
 
 mod probe;
-
-/// Fake DNS TLD used in tests for an invalid hostname.
-const DOT_INVALID: &str = ".invalid";
+mod reportgen;
 
 // The various default timeouts for things.
 
-/// The maximum amount of time netcheck will spend gathering a single report.
-const OVERALL_PROBE_TIMEOUT: Duration = Duration::from_secs(5);
-
-/// The maximum amount of time netcheck will spend probing with STUN packets without getting a
-/// reply before switching to HTTP probing, on the assumption that outbound UDP is blocked.
-const STUN_PROBE_TIMEOUT: Duration = Duration::from_secs(3);
-
-/// The maximum amount of time netcheck will spend probing with ICMP packets.
-const ICMP_PROBE_TIMEOUT: Duration = Duration::from_secs(1);
-
-/// The amount of time we wait for a hairpinned packet to come back.
-const HAIRPIN_CHECK_TIMEOUT: Duration = Duration::from_millis(100);
-
 const FULL_REPORT_INTERVAL: Duration = Duration::from_secs(5 * 60);
 
-const ENOUGH_REGIONS: usize = 3;
-
-// Chosen semi-arbitrarily
-const CAPTIVE_PORTAL_DELAY: Duration = Duration::from_millis(200);
-
-/// Timeout for captive portal checks, must be lower than OVERALL_PROBE_TIMEOUT
-const CAPTIVE_PORTAL_TIMEOUT: Duration = Duration::from_secs(2);
+/// The maximum latency of all regions, if none are found yet.
+///
+/// Normally the max latency of all regions is computed, but if we don't yet know any region
+/// latencies we return this as default.  This is the value of the initial STUN probe
+/// delays.  It is only used as time to wait for further latencies to arrive, which *should*
+/// never happen unless there already is at least one latency.  Yet here we are, defining a
+/// default which will never be used.
+const DEFAULT_MAX_LATENCY: Duration = Duration::from_millis(100);
 
 /// A netcheck report.
 ///
-/// Can be obtained by calling [Client::get_report].
+/// Can be obtained by calling [`Client::get_report`].
```

Comment:
It does, but then it does not treat it as a code block.  I once had an in-depth bikeshedding session with someone about whether you should type these backticks or not.  We eventually settled on "do what Rust std does" as the best tie-breaker, and they put code references in backticks.  So since then it's become a habit to me to include the backticks.

:shrug: